File: passenger-service/src/main/java/com/dulfinne/taxi/passengerservice/service/PassengerService.java
package com.dulfinne.taxi.passengerservice.service;

import com.dulfinne.taxi.passengerservice.dto.request.PassengerRequest;
import com.dulfinne.taxi.passengerservice.dto.response.PaginatedResponse;
import com.dulfinne.taxi.passengerservice.dto.response.PassengerResponse;

import java.util.List;

public interface PassengerService {

  PaginatedResponse<PassengerResponse> getAllPassengers(Integer offset, Integer limit, String sortField);

  PassengerResponse getPassengerByUsername(String username);

  PassengerResponse savePassenger(String username, PassengerRequest passengerRequest);

  PassengerResponse updatePassenger(String username, PassengerRequest request);

  void deletePassenger(String username);
}

File: ride-service/src/main/kotlin/com/dulfinne/taxi/rideservice/service/PassengerService.kt
package com.dulfinne.taxi.rideservice.service

import com.dulfinne.taxi.rideservice.client.dto.DriverResponse
import com.dulfinne.taxi.rideservice.dto.request.LocationRequest
import com.dulfinne.taxi.rideservice.dto.request.RatingRequest
import com.dulfinne.taxi.rideservice.dto.response.CountPriceResponse
import com.dulfinne.taxi.rideservice.dto.response.RideResponse
import org.springframework.data.domain.Page

interface PassengerService {
    fun countPrice(request: LocationRequest): CountPriceResponse
    fun createRide(passengerUsername: String, request: LocationRequest): RideResponse
    fun cancelRide(rideId: Long, passengerUsername: String)
    fun rateDriver(rideId: Long, passengerUsername: String, request: RatingRequest)
    fun getAllPassengerRides(passengerUsername: String, offset: Int, limit: Int, sortField: String): Page<RideResponse>
    fun getRideById(passengerUsername: String, rideId: Long): RideResponse
    fun getDriverProfile(passengerUsername: String, rideId: Long): DriverResponse
}